---
title: "R Notebook"
output: html_notebook
---

This notebook outlines the processes for coassembly and differential coverage binning of subsets of geographically related metagenomes.

1. Coassembly of individual asssemblies using contigs > 2,000 bp 
2. read mapping using [bamM](http://ecogenomics.github.io/BamM/), or bowtie2 and samtools
3. 1st binning step [runMetabat.sh](https://bitbucket.org/berkeleylab/metabat)
4. 2nd step binning [MyCC](https://sourceforge.net/projects/sb2nhri/files/MyCC/). Accurate binning of metagenomic contigs via automated clustering sequences using information of genomic signatures and marker genes [reference](https://www.nature.com/articles/srep24175)
5. 3rd binning option, CONCOCT
6. checkM integrated into binning refinement using metawrap
7. [gtdb-tk](https://github.com/Ecogenomics/GTDBTk)
8. identify viral bins
9. indentify eukaryote bins

where possible example scripts, commands and options are given

### Coassemblies

Coassemblies were carried out by combining the contigs > 2000 bp from each individual megahit assembly and coassembling using the Geneious assembler with default settings.

assembled contigs and unused reads were exported from geneious and the coassembled contigs renamed with unique ids.

```{bash}
perl -ane 'if(/^\>/){$a++;print ">NSI.coass_$a\n"}else{print;}' NSI.2000.coAssembly.fasta > NSI.2000.coAssembly.r.fasta
```

Then raw sequences from each sample were mapped against the coassemblies using bamM
(formerly bowtie2 and samtools)

example: NSI raw sequences mapped against the coassembled contigs from NSI. R1 and R2 are arranged in pairs and the entire set of fastq.gz entered on one line.

```{bash}
bamm make -d NSI.2000.coAssembly.r.fasta -c 21940_1_PE_700bp_MM_UNSW_HMJH2BCXX_TCCTGAGC-TCTCTCCG_L001_R1.fastq.gz 21940_1_PE_700bp_MM_UNSW_HMJH2BCXX_TCCTGAGC-TCTCTCCG_L001_R2.fastq.gz ... 35466_1_PE_480bp_MM_AGRF_H7MYTBCX2_AAGAGGCA-TATCCTCT_L002_R1.fastq.gz 35466_1_PE_480bp_MM_AGRF_H7MYTBCX2_AAGAGGCA-TATCCTCT_L002_R2.fastq.gz -t 12

```

This single bamm make generates all of the .bam files containing the read-mapping information. These were given as input to the automated MetaBata script `runMetaBat.sh`


```{bash}
runMetaBat.sh NSI.2000.coAssembly.r.fasta T2-120151216d0.sam.sorted.bam T4-40020151208d0.sam.sorted.bam T5-40020151205d0.sam.sorted.bam ....
```

runMetaBat.sh attempts to produce the best bins without any parameter optimisation input from the user. This 1st step of binning provides one set of bins for comparison with the next step. The main aim of this step is to produce a depth.covergae.txt file which is produced from `jgi_summarize_bam_contig_depths`. e.g. `jgi_summarize_bam_contig_depths --outputDepth depth.txt *.bam`


MyCC generates bins using tetranucleotide frequencies, clusters them using affinity propagation and then corrects them on the basis of sequences harboring marker genes [Lin & Liao, 2016](https://www.nature.com/articles/srep24175).

MyCC was run in a docker container 

```{bash}
MyCC.py NSI.contigs.2000.coass.fasta -a NSI.contigs.2000.coass.fasta.depth.txt -plot
```


example output

root@5fa8385a979f:/home/20181103_0007_4mer_0.7_cov# ls
1_Data                    Cluster.108.fasta  Cluster.128.fasta  Cluster.148.fasta  Cluster.168.fasta  Cluster.31.fasta  Cluster.51.fasta  Cluster.71.fasta  Cluster.91.fasta
2_GetFeature              Cluster.109.fasta  Cluster.129.fasta  Cluster.149.fasta  Cluster.169.fasta  Cluster.32.fasta  Cluster.52.fasta  Cluster.72.fasta  Cluster.92.fasta
2_GetFeature_4mer         Cluster.11.fasta   Cluster.13.fasta   Cluster.15.fasta   Cluster.17.fasta   Cluster.33.fasta  Cluster.53.fasta  Cluster.73.fasta  Cluster.93.fasta
3_GetMatrix               Cluster.110.fasta  Cluster.130.fasta  Cluster.150.fasta  Cluster.170.fasta  Cluster.34.fasta  Cluster.54.fasta  Cluster.74.fasta  Cluster.94.fasta
3_GetMatrix_4mer          Cluster.111.fasta  Cluster.131.fasta  Cluster.151.fasta  Cluster.171.fasta  Cluster.35.fasta  Cluster.55.fasta  Cluster.75.fasta  Cluster.95.fasta
4_CLR_transformation_SNE  Cluster.112.fasta  Cluster.132.fasta  Cluster.152.fasta  Cluster.172.fasta  Cluster.36.fasta  Cluster.56.fasta  Cluster.76.fasta  Cluster.96.fasta
5_ClusterCorrection       Cluster.113.fasta  Cluster.133.fasta  Cluster.153.fasta  Cluster.173.fasta  Cluster.37.fasta  Cluster.57.fasta  Cluster.77.fasta  Cluster.97.fasta
6_ClusterCtg              Cluster.114.fasta  Cluster.134.fasta  Cluster.154.fasta  Cluster.18.fasta   Cluster.38.fasta  Cluster.58.fasta  Cluster.78.fasta  Cluster.98.fasta
6_ClusterCtg_Alias        Cluster.115.fasta  Cluster.135.fasta  Cluster.155.fasta  Cluster.19.fasta   Cluster.39.fasta  Cluster.59.fasta  Cluster.79.fasta  Cluster.99.fasta
7_AllClusters             Cluster.116.fasta  Cluster.136.fasta  Cluster.156.fasta  Cluster.2.fasta    Cluster.4.fasta   Cluster.6.fasta   Cluster.8.fasta   Cluster.summary
7_AllClusters_Alias       Cluster.117.fasta  Cluster.137.fasta  Cluster.157.fasta  Cluster.20.fasta   Cluster.40.fasta  Cluster.60.fasta  Cluster.80.fasta  ClusterCogs.txt
AP.png                    Cluster.118.fasta  Cluster.138.fasta  Cluster.158.fasta  Cluster.21.fasta   Cluster.41.fasta  Cluster.61.fasta  Cluster.81.fasta  MyCluster.coords
Cluster.1.fasta           Cluster.119.fasta  Cluster.139.fasta  Cluster.159.fasta  Cluster.22.fasta   Cluster.42.fasta  Cluster.62.fasta  Cluster.82.fasta  MyCluster.pdf
Cluster.10.fasta          Cluster.12.fasta   Cluster.14.fasta   Cluster.16.fasta   Cluster.23.fasta   Cluster.43.fasta  Cluster.63.fasta  Cluster.83.fasta  log
Cluster.100.fasta         Cluster.120.fasta  Cluster.140.fasta  Cluster.160.fasta  Cluster.24.fasta   Cluster.44.fasta  Cluster.64.fasta  Cluster.84.fasta
Cluster.101.fasta         Cluster.121.fasta  Cluster.141.fasta  Cluster.161.fasta  Cluster.25.fasta   Cluster.45.fasta  Cluster.65.fasta  Cluster.85.fasta
Cluster.102.fasta         Cluster.122.fasta  Cluster.142.fasta  Cluster.162.fasta  Cluster.26.fasta   Cluster.46.fasta  Cluster.66.fasta  Cluster.86.fasta
Cluster.103.fasta         Cluster.123.fasta  Cluster.143.fasta  Cluster.163.fasta  Cluster.27.fasta   Cluster.47.fasta  Cluster.67.fasta  Cluster.87.fasta
Cluster.104.fasta         Cluster.124.fasta  Cluster.144.fasta  Cluster.164.fasta  Cluster.28.fasta   Cluster.48.fasta  Cluster.68.fasta  Cluster.88.fasta
Cluster.105.fasta         Cluster.125.fasta  Cluster.145.fasta  Cluster.165.fasta  Cluster.29.fasta   Cluster.49.fasta  Cluster.69.fasta  Cluster.89.fasta
Cluster.106.fasta         Cluster.126.fasta  Cluster.146.fasta  Cluster.166.fasta  Cluster.3.fasta    Cluster.5.fasta   Cluster.7.fasta   Cluster.9.fasta
Cluster.107.fasta         Cluster.127.fasta  Cluster.147.fasta  Cluster.167.fasta  Cluster.30.fasta   Cluster.50.fasta  Cluster.70.fasta  Cluster.90.fasta

***

The taxonomic composition of the resulting bins was screened using checkM. Bins identified as bacterial or Archaeal were copied to bacterial_bins and analysed using gtdbtk and the result merged into the XXX.bins.summary

```{bash}
gtdbtk classify_wf --genome_dir ./ --out_dir gtdb.out -x fa
```

'Clean bins' were defined as > 50% completion and < 10% contamination

***

strategies for binning viruses and identifying eukaryotic bins to follow.


Concoct and meta-wrap bin refining 

```{bash}

cut_up_fasta.py NSI.2000.coAssembly.r.fasta -c 10000 -o 0 --merge_last -b contigs_10K.bed > NSI.2000.coAssembly.r.contigs_10k.fa

concoct_coverage_table.py contigs_10K.bed bams/*.bam > coverage_table.tsv

concoct --composition_file EAC.contigs.2000.coass.10K.fa --coverage_file coverage_table.tsv -b EAC.concoct/ -t 20

merge_cutup_clustering.py EAC.concoct/clustering_gt1000.csv > EAC.concoct/clustering_merged.csv

extract_fasta_bins.py EAC.contigs.2000.coass.fa EAC.concoct/clustering_merged.csv --output_path EAC.concoct/bins

metawrap bin_refinement -o refined.6 -A EAC.metabat.bins -B EAC.mycc.bins -C EAC.concoct/bins/ -m 128 -t 20
```

